src/scripts/main.js

```diff
@@ -1,5 +1,25 @@
 'use strict';
 
+const field = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
+const fieldWidth = field.clientWidth;
+const fieldHeight = field.clientHeight;
+const spiderWidth = spider.clientWidth;
+const spiderHeight = spider.clientHeight;
+
 document.addEventListener('click', (e) => {
-  // write code here
+  const x = e.clientX;
+  const y = e.clientY;
+
+  const wallRect = field.getBoundingClientRect();
+
+  let newX = x - wallRect.left - spiderWidth / 2;
+  let newY = y - wallRect.top - spiderHeight / 2;
```

Comment:
The calculation of newX and newY assumes that the field element's top-left corner aligns with the viewport's origin. If the field element is positioned elsewhere, consider using the bounding rectangle's top and left properties to adjust these calculations.